moler/runner.py

```diff
@@ -345,13 +346,13 @@ def wait_for(self, connection_observer, connection_observer_future, timeout=None
                     if (future in done) or connection_observer.done():
                         self._cancel_submitted_future(connection_observer, future)
                         return None
-                    timeout = connection_observer.timeout
+                    timeout = connection_observer.timeout + connection_observer.terminating_timeout
                     already_passed = time.time() - start_time
                     remain_time = timeout - already_passed
 
```

Comment:
remain_time = eol_timeout - already_passed.

Reply:
Inside loop call  _wait_for_time_out (remove future cancel and connection_observer.cncel) and after loop call _remove_future